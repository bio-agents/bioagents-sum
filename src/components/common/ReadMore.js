import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

export default class ReadMore extends PureComponent {
  constructor (props) {
    super(props)

    this.state = {
      expanded: false,
    }
  }

  toggleLines = (e) => {
    // Stops <a href=# </a> from scrolling to the top of the page
    e.preventDefault()

    this.setState({
      expanded: !this.state.expanded,
    })
  }

  render () {
    const { more, less, chars, text } = this.props
    const { expanded } = this.state

    const showText = expanded ? text : text.slice(0, chars)

    return (
      <div>
        <span>{showText}</span>
        {text.length > chars &&
          <span>
            {expanded ? ' ' : '... '}
            <a href='#' onClick={this.toggleLines}>{expanded ? less : more}</a>
          </span>
        }
      </div>
    )
  }
}

ReadMore.defaultProps = {
  chars: 50,
  more: 'Show more',
  less: 'Show less',
}

ReadMore.propTypes = {
  more: PropTypes.string,
  less: PropTypes.string,
  chars: PropTypes.number,
}
