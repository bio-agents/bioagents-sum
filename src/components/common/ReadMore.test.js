import React from 'react'
import renderer from 'react-test-renderer'
import ReadMore from './ReadMore'

it('Renders ReadMore correctly', () => {
  const tree = renderer
    .create(
      <ReadMore text='Lorem Ipsum Something Latinische' more='Show more' less='Show less' chars={10} />
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
