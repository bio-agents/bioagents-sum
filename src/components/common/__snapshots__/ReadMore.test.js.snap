exports[`test Renders ReadMore correctly 1`] = `
<div>
  <span>
    Lorem Ipsu
  </span>
  <span>
    ... 
    <a
      href="#"
      onClick={[Function]}>
      Show more
    </a>
  </span>
</div>
`;
